includes/Handlers/Product/Woo_SOR.php

```diff
@@ -81,12 +81,75 @@ public static function update_catalog_item( CatalogObject $catalog_object, \WC_P
 			$item_data->setReportingCategory( $square_category );
 		}
 
-		$catalog_variations = $item_data->getVariations() ?: array();
+		$attributes = $product->get_attributes();
+
+		$product_variation_ids = $product->get_children();
+		$catalog_variations    = $item_data->getVariations() ? $item_data->getVariations() : array();
 
 		// if dealing with a variable product, try and match the variations
 		if ( $product->is_type( 'variable' ) ) {
 
-			$product_variation_ids = $product->get_children();
+			/**
+			 * If there are multiple variations, it must be a considered as Dynamic Options supported product.
+			 * Create/Update and Assign Dynamic Options only if a product
+			 * has multiple attributes OR options already exists in Square.
+			 */
+			if (
+				count( $attributes ) > 1
+			) {
+				$options_ids            = array();
+				$options_data_transient = get_transient( 'wc_square_options_data' );
```

Comment:
I think we should have a fallback mechanism to generate the requested data in case it is not available in the transient storage. If generating the data is too heavy, we may need to consider saving the data in a more reliable location. This is because there is no guarantee that transient data will persist until its expiration time, it might disappear just seconds after being set. More information [here](https://developer.wordpress.org/apis/transients/#overview:~:text=Everyone%20seems%20to%20misunderstand%20how%20transient%20expiration%20works%2C%20so%20the%20long%20and%20short%20of%20it%20is%3A%20transient%20expiration%20times%20are%20a%20maximum%20time.%20There%20is%20no%20minimum%20age.%20Transients%20might%20disappear%20one%20second%20after%20you%20set%20them%2C%20or%2024%20hours%2C%20but%20they%20will%20never%20be%20around%20after%20the%20expiration%20time.)